internal/mock/cloudsql.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2021 Google LLC
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     https://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mock
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"math/big"
+	"time"
+)
+
+// CloudSQLInst represents settings for a specific Cloud SQL instance.
+//
+// Use NewCloudSQLInstance to instantiate.
+type CloudSQLInst struct {
+	project string
+	region  string
+	name    string
+
+	dbVersion string
+
+	privKey *rsa.PrivateKey
+	cert    *x509.Certificate
+}
+
+// NewCloudSQLInst returns a CloudSQLInst object for configuring mocks.
+func NewCloudSQLInst(project, region, name string) (CloudSQLInst, error) {
+	// TODO: consider options for this?
+	privKey, cert, err := generateInstanceCerts()
+	if err != nil {
+		return CloudSQLInst{}, err
+	}
+
+	c := CloudSQLInst{
+		project:   project,
+		region:    region,
+		name:      name,
+		dbVersion: "POSTGRES_12", // default of no particular importance
```

Comment:
If this dbVersion doesn't matter, shall we just hard-code an empty string in the mock responses?

Reply:
It doesn't matter yet, it almost certainly will at some point in the (near) future, since certain behavior is database specific (like unix socket path). I started fleshing out full options for customizing this, but stopped since I didn't actually need anything for the existing tests.